### PR TITLE
AGPUSH-1305 : add an apns section to the push message API

### PR DESCRIPTION
**Pull request by sebastienblanc**
Like we did for Windows, moving specific payload for iOS/Safari to its own section
@edewit @corinnekrych @cvasilak You mind to review ?

### CODE REVIEW

File: push/sender/src/main/java/org/jboss/aerogear/unifiedpush/message/sender/APNsPushNotificationSender.java [hunk 1]
```diff
@@ -76,15 +76,15 @@ public void sendPushMessage(final Variant variant, final Collection<String> toke
         PayloadBuilder builder = APNS.newPayload()
                 // adding recognized key values
                 .alertBody(message.getAlert()) // alert dialog, in iOS or Safari
-                .alertTitle(message.getTitle()) // The title of the notification in Safari
-                .alertAction(message.getAction()) // The label of the action button, if the user sets the notifications to appear as alerts in Safari.
-                .urlArgs(message.getUrlArgs())
                 .badge(message.getBadge()) // little badge icon update;
                 .sound(message.getSound()) // sound to be played by app
-                .category(message.getActionCategory()); // iOS8: User Action category
+                .alertTitle(message.getApns().getTitle()) // The title of the notification in Safari
```
Comment (matzew) [inline push/sender/src/main/java/org/jboss/aerogear/unifiedpush/message/sender/APNsPushNotificationSender.java]: could we not just stash the APns objc?

File: push/model/src/test/resources/message-format.json [hunk 2]
```diff
@@ -20,9 +18,14 @@
       "textFields": [
       ]
     },
-    "action-category": "some value",
-    "url-args": null,
```
Comment (matzew) [inline push/model/src/test/resources/message-format.json]: is this testing the magic transformation thing from @edewit ?

Comment (edewit) [inline push/model/src/test/resources/message-format.json]: nope this is used in the serializer test

**Comment by edewit** [comment]
@sebastienblanc you'll need to update jaxrs/src/main/resources/rest/sender/100.json so that the old api doesn't break

**Comment by sebastienblanc** [comment]
@edewit Good catch ! But that means we need some tests for that since it did not break the build for me ;)

**Comment by edewit** [comment]
We decided that we would keep the old api usable and transform that to the new api so creating a test that would break if we modify the existing api would be hard as we also add new things.

**Comment by sebastienblanc** [comment]
@edewit Could you check the latest commit, it breaks the requestTransformer unit test, not sure why but it puts the "apns" section inside "message/user-data" instead of just "message/": 
```
 RequestTransformerTest.shouldTransformSenderRequest:48 expected:<{"message":{"alert":"HELLO!","sound":"default","badge":2,"user-data":{"key":"value","key2":"other value"},"apns":{"content-available":true,"action-category":"some value"},"simple-push":"version=123"},"criteria":{"alias":["someUsername"],"deviceType":["someDevice"],"categories":["someCategories"],"variants":["someVariantIDs"]},"config":{"ttl":3600}}> but was:<{"message":{"alert":"HELLO!","sound":"default","badge":2,"user-data":{"key":"value","key2":"other value","apns":{"action-category":"some value","content-available":true}},"simple-push":"version=123"},"criteria":{"alias":["someUsername"],"deviceType":["someDevice"],"categories":["someCategories"],"variants":["someVariantIDs"]},"config":{"ttl":3600}}>

```

**Comment by matzew** [comment]
@sebastienblanc can we merge this ?

**Comment by sebastienblanc** [comment]
@matzew IMO yes

**Comment by sebastienblanc** [comment]
@matzew Hum, now I'm looking , let me handle https://issues.jboss.org/browse/AGPUSH-1259 also in this PR.

**Comment by matzew** [comment]
@sebastienblanc you mean AGPUSH-1304, right ?

**Comment by sebastienblanc** [comment]
@matzew correct :) sorry

File: push/sender/src/main/java/org/jboss/aerogear/unifiedpush/message/sender/APNsPushNotificationSender.java [hunk 3]
```diff
@@ -73,23 +74,30 @@ public void sendPushMessage(final Variant variant, final Collection<String> toke
...
         PayloadBuilder builder = APNS.newPayload()
                 // adding recognized key values
                 .alertBody(message.getAlert()) // alert dialog, in iOS or Safari
-                .alertTitle(message.getTitle()) // The title of the notification in Safari
-                .alertAction(message.getAction()) // The label of the action button, if the user sets the notifications to appear as alerts in Safari.
-                .urlArgs(message.getUrlArgs())
                 .badge(message.getBadge()) // little badge icon update;
                 .sound(message.getSound()) // sound to be played by app
-                .category(message.getActionCategory()); // iOS8: User Action category
+                .alertTitle(apns.getTitle()) // The title of the notification in Safari
```
Comment (matzew) [inline push/sender/src/main/java/org/jboss/aerogear/unifiedpush/message/sender/APNsPushNotificationSender.java]: it's not just for Safari - also for Apple Watch

**Comment by matzew** [comment]
landed on master :tada:

**Comment by matzew** [comment]
:tanabata_tree:

**Pull request closed without merging**